File: WorldFeed/Server/src/WorldFeed.Common/Models/TrendLocation.cs
namespace WorldFeed.Common.Models
{
    using Newtonsoft.Json;

    using WorldFeed.Common.Public.Models.Enums;
    using WorldFeed.Common.Public.Models.Interfaces;

    public class TrendLocation : ITrendLocation
    {
        private class PlaceTypeDTO
        {
            [JsonProperty("name")]
            public string Name { get; set; }

            [JsonProperty("code")]
            public int Code { get; set; }
        }

        [JsonProperty("woeid")]
        public long WoeId { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("country")]
        public string Country { get; set; }

        [JsonProperty("countryCode")]
        public string CountryCode { get; set; }

        [JsonProperty("url")]
        public string Url { get; set; }

        [JsonProperty("parentid")]
        public long ParentId { get; set; }

        [JsonProperty]
        public PlaceType PlaceType
        {
            get
            {
                if (this.placeTypeDTO == null)
                {
                    return PlaceType.Undefined;
                }

                return (PlaceType)this.placeTypeDTO.Code;
            }
            set
            {
                if (this.placeTypeDTO == null)
                {
                    this.placeTypeDTO = new PlaceTypeDTO();
                }

                this.placeTypeDTO.Code = (int) value;
            }
        }

        [JsonProperty("placeType")]
        private PlaceTypeDTO placeTypeDTO { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/Models/TrendLocation.ts
import {ITrendLocation} from "../../Public/Models/Interfaces/ITrendLocation";
import {PlaceType} from '../../Public/Models/Enum/PlaceType';

export class TrendLocation implements ITrendLocation {
  private PlaceTypeDTO = class PlaceTypeDTO {
    // [JsonProperty("name")]
    public name: string;

    // [JsonProperty("code")]
    public code: number;
  };

  // [JsonProperty("woeid")]
  public woeId: number;

  // [JsonProperty("name")]
  public name: string;

  // [JsonProperty("country")]
  public country: string;

  // [JsonProperty("countryCode")]
  public countryCode: string;

  // [JsonProperty("url")]
  public url: string;

  // [JsonProperty("parentid")]
  public parentId: number;

  // [JsonProperty]
  get placeType(): PlaceType {
    if (this._placeTypeDTO == null) {
      return PlaceType.Undefined;
    }

    return this._placeTypeDTO.Code as PlaceType;
  }

  set placeType(value: PlaceType) {
    if (this._placeTypeDTO == null) {
      this._placeTypeDTO = new this.PlaceTypeDTO();
    }

    this._placeTypeDTO.Code = value as number;
  }

  // [JsonProperty("placeType")]
  private _placeTypeDTO: any; // PlaceTypeDTO;
}
